src/VisualStudio/Core/Def/Storage/AbstractCloudCachePersistentStorageService.cs

```diff
@@ -40,13 +40,16 @@ protected sealed override bool ShouldDeleteDatabase(Exception exception)
         protected sealed override async ValueTask<IChecksummedPersistentStorage?> TryOpenDatabaseAsync(
             SolutionKey solutionKey, string workingFolderPath, string databaseFilePath, CancellationToken cancellationToken)
         {
-            var cacheService = await this.CreateCacheServiceAsync(cancellationToken).ConfigureAwait(false);
+            var solutionFolder = IOUtilities.PerformIO(() => Path.GetDirectoryName(solutionKey.FilePath));
+            if (solutionFolder == null || solutionFolder == "")
+                return null;
```

Comment:
string.IsNullOrEmpty didn't work as this is linked into projects whose string apis are not nrt annotated.

Reply:
Don't we still have RoslynString.IsNullOrEmpty?

Reply:
ooh?